qiskit_ibm_runtime/visualization/draw_zne.py

```diff
@@ -0,0 +1,400 @@
+                continue
+            fig.add_traces(
+                [
+                    _scatter_trace(
+                        noise_factors,
+                        result.data.evs_noise_factors[idx],
+                        result.data.stds_noise_factors[idx],
+                        names[i],
+                        i,
+                        color,
+                        show_legend,
+                    ),
+                    *_line_fill_trace(
+                        e_noise_factors,
+                        evs,
+                        stds,
+                        n_stds,
+                        name=extrapolator,
+                        legend_group=i,
+                        color=color,
+                    ),
+                ],
+                rows=1,
+                cols=idx_m + 1,
+            )
+            show_legend = False
+
+    fig.update_layout(
+        height=height,
+        width=width,
+        plot_bgcolor="white",
+        hoverlabel_align="right",
+    )
+
+    fig.update_xaxes(
+        mirror=True,
+        ticks="outside",
+        showline=True,
+        linecolor="black",
+        gridcolor="lightgrey",
+    )
+
+    fig.update_yaxes(
+        range=[
+            np.min(result.data.evs_noise_factors) - max_std,
+            np.max(result.data.evs_noise_factors) + max_std,
+        ],
+        mirror=True,
+        ticks="outside",
+        showline=True,
+        linecolor="black",
+        gridcolor="lightgrey",
+    )
+
+    return fig
+
+
+def _line_fill_trace(
+    x_values: np.array,
+    y_values: np.array,
+    stds: np.array,
+    n_stds: int = 1,
+    name: str | None = None,
+    legend_group: int | None = None,
+    color: str | None = None,
+    show_legend: bool = False,
+) -> list[PlotlyScatter]:
+    """Return a list of traces for a line plot with a standard deviation fill.
+
+    Args:
+        x_values: The values for the x-axis.
+        y_values: The values for the y-axis.
+        stds: The standard deviation for the ``y_values``.
+        n_stds: The number of standard deviations to include around the line plot.
+        name: The name of this trace.
+        legend_group: The legend group that this trace belongs to.
+        color: The color of the fill around the line.
+        show_legend: Whether to show this trace on a figure.
+
+    Returns:
+        A list of traces.
+    """
+    go = plotly_module(".graph_objects")
+    hovertemplate = "extrapolator: " + name + "<br>noise_factor: %{x}<br>ev_extrap: %{y:.4f}"
+    hovertemplate += "<br>std: %{customdata:.4f}<extra></extra>"
+    return [
+        go.Scatter(
+            x=x_values,
+            y=y_values,
+            customdata=stds,
+            name=name,
+            mode="lines",
+            line={"color": color},
+            hovertemplate=hovertemplate,
+            legendgroup=legend_group,
+            showlegend=show_legend,
+        ),
+        *(
+            go.Scatter(
```

Comment:
If possible, replace this with a single scatter trace using `None`s as discussed privately

Reply:
Attempted the above, did not affect the opacity as expected. Marking as resolved.